packages/o-spreadsheet-engine/src/plugins/ui_stateful/clipboard.ts

```diff
@@ -674,9 +675,14 @@ export class ClipboardPlugin extends UIPlugin {
 
   private getClipboardData(zones: Zone[]): ClipboardData {
     const sheetId = this.getters.getActiveSheetId();
-    const selectedFigureId = this.getters.getSelectedFigureId();
-    if (selectedFigureId) {
-      return { figureId: selectedFigureId, sheetId };
+    const selectedFiguresIds = this.getters.getSelectedFiguresIds();
+    const figuresUI = selectedFiguresIds
+      .map((id) => this.getters.getFigure(sheetId, id))
+      .filter(isDefined)
+      .map((f) => this.getters.getFigureUI(sheetId, f));
+    const topLeft = rectUnion(...figuresUI);
```

Comment:
I'd say it's the responsibility of the clipboard handlers to figure the top-left, not of the clipboard plugin